src/Stratis.Bitcoin/Consensus/Validators/BlockValidator.cs

```diff
@@ -59,8 +62,9 @@ public interface IIntegrityValidator
         /// </remarks>
         /// <param name="header">The chained header that is going to be validated.</param>
         /// <param name="block">The block that is going to be validated.</param>
+        /// <param name="blockMined">Whether the block was mined or obtained from a peer.</param>
         /// <returns>Context that contains validation result related information.</returns>
-        ValidationContext VerifyBlockIntegrity(ChainedHeader header, Block block);
+        ValidationContext VerifyBlockIntegrity(ChainedHeader header, Block block, bool blockMined);
```

Comment:
Do we need this?

Reply:
yeah. there is one integration rule that uses context.BlockMined